lib/trino-orc/src/main/java/io/trino/orc/TupleDomainOrcPredicate.java

```diff
@@ -145,7 +147,7 @@ private boolean columnOverlaps(Domain predicateDomain, long numberOfRows, Column
     private static Optional<Collection<Object>> extractDiscreteValues(ValueSet valueSet)
     {
         if (!valueSet.isDiscreteSet()) {
-            return Optional.empty();
+            return tryExpandRanges(valueSet, DEFAULT_COMPACTION_THRESHOLD);
```

Comment:
Instead of the hard-coded value, let's use the value of compaction threshold that comes from config/session property.

Reply:
Sounds good -> `HiveConfig#getDomainCompactionThreshold`